src/firmware_new/boards/frankie_v1/freertos.c

```diff
@@ -47,15 +52,191 @@
 /* USER CODE BEGIN FunctionPrototypes */
 
 /* USER CODE END FunctionPrototypes */
 
+void io_proto_multicast_startNetworkingTask(void *argument);
+extern void io_proto_multicast_sender_task(void *argument);
+extern void io_proto_multicast_listener_task(void *argument);
+void test_msg_update(void *argument);
+
+extern void MX_LWIP_Init(void);
+void MX_FREERTOS_Init(void); /* (MISRA C 2004 rule 8.1) */
+
+/**
+ * @brief  FreeRTOS initialization
+ * @param  None
+ * @retval None
+ */
+void MX_FREERTOS_Init(void)
+{
+    /* USER CODE BEGIN Init */
+
+    /* USER CODE END Init */
+
+    /* USER CODE BEGIN RTOS_MUTEX */
+    /* add mutexes, ... */
+    /* USER CODE END RTOS_MUTEX */
+
+    /* USER CODE BEGIN RTOS_SEMAPHORES */
+    /* add semaphores, ... */
+    /* USER CODE END RTOS_SEMAPHORES */
+
+    /* USER CODE BEGIN RTOS_TIMERS */
+    /* start timers, add new ones, ... */
+    /* USER CODE END RTOS_TIMERS */
+
+    /* USER CODE BEGIN RTOS_QUEUES */
+    /* add queues, ... */
+    /* USER CODE END RTOS_QUEUES */
+
+    /* Create the thread(s) */
+    /* creation of NetStartTask */
+    NetStartTaskHandle = osThreadNew(io_proto_multicast_startNetworkingTask, NULL,
+                                     &NetStartTask_attributes);
+
+    /* creation of RobotStatusTask */
+    RobotStatusTaskHandle =
+        osThreadNew(io_proto_multicast_sender_task,
+                    (void *)tbots_robot_msg_sender_profile, &RobotStatusTask_attributes);
+
+    /* creation of VisionMsgTask */
+    VisionMsgTaskHandle =
+        osThreadNew(io_proto_multicast_listener_task, (void *)vision_msg_listener_profile,
+                    &VisionMsgTask_attributes);
+
+    /* creation of PrimMsgTask */
+    PrimMsgTaskHandle =
+        osThreadNew(io_proto_multicast_listener_task,
+                    (void *)primitive_msg_listener_profile, &PrimMsgTask_attributes);
+
+    /* creation of testMsgUpdate */
+    testMsgUpdateHandle =
+        osThreadNew(test_msg_update, (void *)tbots_robot_msg_sender_profile,
+                    &testMsgUpdate_attributes);
+
+    /* USER CODE BEGIN RTOS_THREADS */
+    /* add threads, ... */
+    /* USER CODE END RTOS_THREADS */
+}
+
+/* USER CODE BEGIN Header_io_proto_multicast_startNetworkingTask */
+/**
+ * @brief  Function implementing the NetStartTask thread.
+ * @param  argument: Not used
+ * @retval None
+ */
+/* USER CODE END Header_io_proto_multicast_startNetworkingTask */
+__weak void io_proto_multicast_startNetworkingTask(void *argument)
+{
+    /* init code for LWIP */
+    MX_LWIP_Init();
+    /* USER CODE BEGIN io_proto_multicast_startNetworkingTask */
+    /* Infinite loop */
+    for (;;)
+    {
+        osDelay(1);
+    }
+    /* USER CODE END io_proto_multicast_startNetworkingTask */
+}
+
+/* USER CODE BEGIN Header_test_msg_update */
+/**
+ * @brief Function implementing the testMsgUpdate thread.
+ * @param argument: Not used
+ * @retval None
```

Comment:
Just checking standard practice if there's any need to comment `retval` if it's void.

Reply:
this is an autogenerated javadoc which we have no control over, its far from the standard, but its what cube gives us

Reply:
Are we sure this is autogenerated? Looks like it's inside `USER CODE` blocks?

Reply:
Ahh yes you are right, I tried this before and I _think_ in the previous cube version used to wipe it, but they may have fixed it (or maybe I did something wrong)

Its probably fine to leave as-is here though, because its a test task that will get wiped soon :tm: